tests/helpers/test_helpers.go

```diff
@@ -0,0 +1,210 @@
+package helpers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/postgres" // Initializing Postgres driver
+	_ "github.com/golang-migrate/migrate/v4/source/file"       // Initializing file source driver
+	"github.com/ihorlenko/weather_notifier/internal/app"
+	"github.com/ihorlenko/weather_notifier/internal/config"
+	"github.com/ihorlenko/weather_notifier/internal/database"
+	"github.com/ihorlenko/weather_notifier/internal/server"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"gorm.io/gorm"
+)
+
+type TestSuite struct {
+	PgContainer  testcontainers.Container
+	DB           *gorm.DB
+	Config       *config.Config
+	Router       *gin.Engine
+	TestServer   *httptest.Server
+	AppContainer *app.Container
+	ctx          context.Context
+}
+
+func SetupTestSuite(t *testing.T) *TestSuite {
+	ctx := context.Background()
+
+	projectRoot, err := findProjectRoot()
+	require.NoError(t, err)
+
+	pgContainer, err := postgres.Run(ctx,
+		"postgres:15-alpine",
+		postgres.WithDatabase("test_weather_api"),
+		postgres.WithUsername("test_user"),
+		postgres.WithPassword("test_password"),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").
+				WithOccurrence(2).
+				WithStartupTimeout(30*time.Second)),
+	)
+	require.NoError(t, err)
+
+	host, err := pgContainer.Host(ctx)
+	require.NoError(t, err)
+
+	port, err := pgContainer.MappedPort(ctx, "5432")
+	require.NoError(t, err)
+
+	cfg := &config.Config{
+		AppConfig: config.AppConfig{
+			BaseURL: "http://localhost:8080",
+			Port:    "8080",
+		},
+		DBConfig: config.DBConfig{
+			Host:     host,
+			Port:     port.Port(),
+			User:     "test_user",
+			Password: "test_password",
+			DBName:   "test_weather_api",
+			SSLMode:  "disable",
+		},
+		WeatherAPIConfig: config.WeatherAPIConfig{
+			APIKey: "test_api_key", // We'll mock this
+		},
+		EmailConfig: config.EmailConfig{
+			From:     "test@example.com",
+			Password: "test_password",
+			SMTPHost: "localhost",
+			SMTPPort: "587",
+		},
+	}
+
+	os.Setenv("TEST_MIGRATIONS_PATH", filepath.Join(projectRoot, "migrations"))
+	defer os.Unsetenv("TEST_MIGRATIONS_PATH")
+
+	err = runTestMigrations(cfg)
```

Comment:
not a call to action
wonder what will happen when your db has 50+ migrations and you will have 100+ tests. Not sure if it's a good practice. 
If you do not have separate stage for tests may be just create separate test migration or test-ready sqllite if possible